okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpChannelBuilder.java

```diff
@@ -117,15 +118,30 @@ public OkHttpChannelBuilder transportExecutor(ExecutorService executor) {
    * <p>Should only used by tests.
    */
   public OkHttpChannelBuilder overrideHostForAuthority(String host) {
-    this.authorityHost = host;
+    this.authorityHost = Preconditions.checkNotNull(host);
+    return this;
+  }
+
+  /**
+   * Sets the negotiation type for the HTTP/2 connection.
+   *
+   * <p>Default: <code>TLS</code>
+   */
+  public OkHttpChannelBuilder negotiationType(NegotiationType type) {
+    negotiationType = Preconditions.checkNotNull(type);
     return this;
   }
 
   /**
-   * Provides a SSLSocketFactory to establish a secure connection.
+   * Provides a SSLSocketFactory to replace the default SSLSocketFactory used for TLS.
+   *
+   * <p>By default, when TLS is enabled, <code>SSLSocketFactory.getDefault()</code> will be used.
+   *
+   * <p><code>negotiationType(NegotiationType.TLS)</code> will be called inside this method.
```

Comment:
You say it will be called, but you explaining it from the user's stand point. How about:

> The {@link #negotationType} will be set to {@link NegotiationType#TLS}.